src/ProcessStamp.php

```diff
@@ -55,31 +56,33 @@ public static function firstOrCreateByProcess(array $process, ?string $hash = nu
             $hash = static::makeProcessHash($process);
         }
 
-        $parent = null;
+        return retry(4, function() use ($hash, $process) {
+            $parent = null;
 
-        if (config('process-stamps.resolve_recursive') && ! empty($process['parent_name'])) {
-            $parent = static::firstOrCreateByProcess(static::getProcessName($process['type'], $process['parent_name']));
-        }
+            if (config('process-stamps.resolve_recursive') && ! empty($process['parent_name'])) {
+                $parent = static::firstOrCreateByProcess(static::getProcessName($process['type'], $process['parent_name']));
+            }
```

Comment:
We should restrucutre this, blah blah:

```suggestion
            if (! config('process-stamps.resolve_recursive') && empty($process['parent_name'])) {
                $parent = static::firstOrCreateByProcess(static::getProcessName($process['type'], $process['parent_name']));
            }
```